src/inventory_ui.cpp

```diff
@@ -834,110 +834,103 @@ int game::inv_for_salvage(const std::string &title, const salvage_actor& actor )
-                    stacks = true;
-                    elem.push_back( *candidate );
-                    break;
-                }
-            }
-
-            if( !stacks ) {
-                item_stacks.push_back( std::list<item>( 1, *candidate ) );
-
-                if( cur_invlet <= last_invlet ) {
-                    item_stacks.back().front().invlet = cur_invlet;
-                    cur_invlet++;
-                } else {
-                    item_stacks.back().front().invlet = ' ';
-                }
-
-                selectables.push_back( &*candidate );
-            }
-        }
-    }
-
-    for( size_t a = 0; a < item_stacks.size(); a++ ) {
-        // avoid INT_MIN, as it can be confused with "no item at all"
-        result_slice.push_back( indexed_invslice::value_type( &item_stacks[a], INT_MIN + a + 1 ) );
-    }
-}
-
-item_location game::inv_map_splice( item_filter filter, const std::string &title )
-{
-    return inv_map_splice( filter, filter, filter, title );
+    return inv_map_splice( filter, filter, filter, title, radius );
 }
 
 item_location game::inv_map_splice(
-    item_filter inv_filter, item_filter ground_filter, item_filter vehicle_filter, const std::string &title )
+    item_filter inv_filter, item_filter ground_filter, item_filter vehicle_filter, const std::string &title, int radius )
 {
-    char cur_invlet = '0';
-
-    pseudo_inventory ground_items;
-    pseudo_inventory vehicle_items;
-
-    std::vector<item *> ground_selectables;
-    std::vector<item *> vehicle_selectables;
-
-    indexed_invslice ground_items_slice;
-    indexed_invslice veh_items_slice;
+    inventory_selector inv_s(false, false, title);
 
-    if( !m.has_flag( "SEALED", g->u.pos() ) ) {
-        pseudo_inv_to_slice( m.i_at( g->u.pos() ), ground_filter,
-                             ground_items, ground_items_slice,
-                             ground_selectables, cur_invlet );
-    }
+    // first get matching items from the inventory
+    u.inv.restack(&u);
+    u.inv.sort();
+    inv_s.make_item_list(u.inv.slice_filter_by(inv_filter));
+
+    static const item_category ground_cat ("GROUND:",  _("GROUND:"),  -1000);
+    static const item_category nearby_cat ("NEARBY:",  _("NEARBY:"),  -2000);
+    static const item_category vehicle_cat("VEHICLE:", _("VEHICLE:"), -3000);
+
+    std::vector<std::vector<std::list<item>>> stacks;
+    std::vector<indexed_invslice> slices;
+    std::unordered_map<item *, item_location> opts;
+
+    const char min_invlet = '0';
+    const char max_invlet = '9';
+    char cur_invlet = min_invlet;
+    std::vector<item_location> invlets;
+
+    for (const auto& pos : closest_tripoints_first(radius, g->u.pos())) {
+        // second get all matching items on the map within radius
+        if (m.accessible_items(g->u.pos(), pos, radius)) {
+            auto items = m.i_at(pos);
+
+            // create a new slice and stack for the current map tile
+            stacks.emplace_back(items.size());
```

Comment:
I'm not sure what this is trying to do, it inserts a new vector into stacks that contains items.size() default constructed lists, which means they are empty lists, and they will cause a segfault when list::back() is called in the find_if() that is trying to match them up.  Did you want a vector with items.size() capacity?
Obviously the vehicle section has the same issue.